mission_systems/shooter/firmware/shooter.cpp

```diff
@@ -223,46 +351,56 @@ class Comms
       feeder.set(req.feeder);
       shooter.set(req.shooter);
     }
-    //~ static void messageCallback(const std_msgs::String& str_msg)
-    //~ {
-      //~ String s = str_msg.data;
-      //~ if (s == "flyon")
-        //~ shooter.on();
-      //~ else if (s == "flyoff")
-        //~ shooter.off();
-      //~ else if (s == "feedon")
-        //~ feeder.on();
-      //~ else if (s == "feedoff")
-        //~ feeder.off();
-      //~ else if (s == "feedreverse")
-        //~ feeder.reverse();
-      //~ else if (s == "shoot")
-        //~ autoController.shoot();
-      //~ else if (s == "cancel")
-        //~ autoController.cancel();
-      //~ else if (s == "ledon")
-        //~ digitalWrite(13,HIGH);
-      //~ else if (s == "ledoff")
-        //~ digitalWrite(13,LOW);
-    //~ }
+    #else
+    static void fireCallback(const std_srvs::Trigger::Request &req, std_srvs::Trigger::Response &res)
+    {
+      autoController.shoot();
+      res.success = true;
+    }
+    static void cancelCallback(const std_srvs::Trigger::Request &req, std_srvs::Trigger::Response &res)
+    {
+      autoController.cancel();
+      res.success = true;
+    }
+    static void manualCallback(const navigator_msgs::ShooterManual::Request &req, navigator_msgs::ShooterManual::Response &res)
+    {
+      autoController.cancel();
+      feeder.set(req.feeder);
+      shooter.set(req.shooter);
+      res.success = true;
+    }    
+    #endif
+
   public:
     Comms() :
-      str_msg(),
-      //~ sub("/shooter/control",&messageCallback),
+      #ifdef USE_LINEAR_FEEDER
+      fireService("/shooter/fire", &fireCallback),
```

Comment:
Can you put comments in here as to what does what? One time, Daniel and I spent 10 or 15 minutes trying to figure out how to use all the service calls when we were testing.